scripts/top-search-queries/get-top-search-results.go

```diff
@@ -0,0 +1,243 @@
+// written to the supplied output CSV file as separate rows.
+func getQueryResults(ctx context.Context, querySupplied, nlpWeighting string, cfg runConfig) searchResponse {
+	sdkItems := callSearchAPI(ctx, querySupplied, nlpWeighting, cfg.APIURL)
+
+	searchItems := make([]searchItem, 0)
+	for i := range sdkItems {
+		var searchItem searchItem
+		searchItem.Uri = sdkItems[i].URI
+		searchItem.Edition = sdkItems[i].Edition
+		searchItem.Title = sdkItems[i].Title
+		searchItem.Type = sdkItems[i].DataType
+		searchItem.Summary = sdkItems[i].Summary
+		searchItem.ReleaseDate = sdkItems[i].ReleaseDate
+		searchItems = append(searchItems, searchItem)
+	}
+
+	var responseObject searchResponse
+	responseObject.Items = searchItems
+	return responseObject
+}
+
+// AddResultsToCSV writes the results of a Search API query call to the supplied output CSV file (as separate rows).
+func AddResultsToCSV(ctx context.Context, querySupplied string, csvFile *os.File, nlpWeighting string, responseObject searchResponse) {
+	now := time.Now()
+	dateTimeRequest := now.Format(time.DateTime)
+	nlpOnOrOff := "off"
+	if nlpWeighting == "true" {
+		nlpOnOrOff = "on"
+	}
+	csvWriter := csv.NewWriter(csvFile)
+	defer csvWriter.Flush()
+
+	// The number of items in the responseObject will be 10 or fewer (as 10 was specified as the limit in the query)
+	for i := 0; i < len(responseObject.Items); i++ {
+		item := responseObject.Items[i]
+		addItemToCSV(ctx, querySupplied, nlpOnOrOff, dateTimeRequest, item, i, csvWriter)
+	}
+}
+
+// addItemToCSV adds a row of data, taken from one item of the results (of a particular query), to the output CSV file.
+func addItemToCSV(ctx context.Context, querySupplied, nlpOnOrOff, dateTimeRequest string, item searchItem, position int, csvWriter *csv.Writer) {
+	rowNum := position + 1
+	resultsRow := make([]string, 10)
+	resultsRow[0] = nlpOnOrOff
+	resultsRow[1] = dateTimeRequest
+	resultsRow[2] = querySupplied
+	resultsRow[3] = item.ReleaseDate
+	resultsRow[4] = item.Type
+	resultsRow[5] = strconv.Itoa(rowNum)
+	resultsRow[6] = item.Title
+	resultsRow[7] = item.Uri
+	resultsRow[8] = item.Edition
+	resultsRow[9] = item.Summary
+
+	err := csvWriter.Write(resultsRow)
+	check(ctx, fmt.Sprintf("failed writing results row for query '%s' at row %d", querySupplied, rowNum), err)
+}
+
+// callSearchAPI calls the live Search API using the supplied values of query and nlpWeighting for the relevant query
+// parameters. It specifies a limit of 10 results to be returned in the query response.
+func callSearchAPI(ctx context.Context, query, nlpWeighting, apiURL string) []models.Item {
+	urlStr := fmt.Sprintf("%s/search?q=%s&limit=10&nlp_weighting=%s", apiURL, query, nlpWeighting)
+	logData := log.Data{"query to construct: ": urlStr}
+	log.Info(ctx, "using SDK to call Search API", logData)
+
+	queryVals := url.Values{}
+	queryVals.Add("q", query)
+	queryVals.Add("limit", "10")
```

Comment:
Probably shouldn't hard code 10, best to make it another command line arg that can be defaulted to 10